server/ec2spotmanager/cron.py

```diff
@@ -34,10 +34,10 @@ def update_stats():
             current_delta_entry = PoolUptimeDetailedEntry()
             current_delta_entry.pool = pool
 
-        current_delta_entry.target = pool.config.flatten().size
+        current_delta_entry.target = pool.config.provider.flatten().size
 
-        actual = Instance.objects.filter(pool=pool).filter(Q(status_code=INSTANCE_STATE['pending']) |
-                                                           Q(status_code=INSTANCE_STATE['running'])).count()
+        actual = EC2Instance.objects.filter(pool=pool).filter(Q(status_code=EC2_INSTANCE_STATE['pending']) |
+                                                           Q(status_code=EC2_INSTANCE_STATE['running'])).count()
```

Comment:
This should later probably be abstracted in such a way that the specific instance model decides when machines are live/pending etc., and this code calls the abstraction for that. The notion of pending, running, etc is not EC2 specific.